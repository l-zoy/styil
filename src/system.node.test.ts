/**
 * @jest-environment node
 */
import {
  createSystem,
  createExtracts as defaultCreateExtracts,
  styled as defaultStyled,
  flush as defaultFlush
} from './system'

describe('css node', () => {
  it('createStyil node', () => {
    const { createExtracts, styled, flush } = createSystem()

    styled('div', {
      width: 100
    })

    styled('div', {
      width: 100
    })

    styled('div', {
      height: 100
    })

    expect(createExtracts().extractHtml).toMatchSnapshot()

    flush()
  })

  it('node default createExtracts ', () => {
    defaultStyled('div', {
      width: 100
    })

    defaultStyled('div', {
      width: 100
    })

    defaultStyled('div', {
      height: 100
    })

    expect(defaultCreateExtracts().extractHtml).toMatchSnapshot()

    defaultFlush()
  })
})
