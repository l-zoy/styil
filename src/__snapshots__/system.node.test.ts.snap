// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css node createStyil node 1`] = `
"
      <meta id=\\"__styil_cache__\\" name=\\"styil-cache\\" mode=\\"none\\" content=\\"css-1473887168|css-2950234073\\">
      <style data-styil=\\"css\\">.css-1473887168{width:100px;}.css-2950234073{height:100px;}</style>
    "
`;

exports[`css node node default getCssValue  1`] = `
"
      <meta id=\\"__styil_cache__\\" name=\\"styil-cache\\" mode=\\"none\\" content=\\"css-1473887168|css-2950234073\\">
      <style data-styil=\\"css\\">.css-1473887168{width:100px;}.css-2950234073{height:100px;}</style>
    "
`;
